### PR TITLE
refactor get set tagging apis for bucket and object

### PR DESCRIPTION
**Pull request by prakashsvmx**
refactor get set tagging apis for bucket and object

### CODE REVIEW

File: src/internal/client.ts [hunk 1]
```diff
@@ -1626,4 +1624,110 @@ export class TypedClient {
...
+        TagSet: {
+          Tag: tagsList,
+        },
+      },
+    }
+    const encoder = new TextEncoder()
+    const headers = {} as RequestHeaders
+    const builder = new xml2js.Builder({ headless: true, renderOpts: { pretty: false } })
+    const payload = builder.buildObject(taggingConfig)
+    const payloadBuf = Buffer.from(encoder.encode(payload))
```
Comment (trim21) [inline src/internal/client.ts:1651]: same with last comment, `Buffer.from(payload)` should work fine.

File: src/internal/client.ts [hunk 2]
```diff
@@ -1626,4 +1624,110 @@ export class TypedClient {
...
+      tagsList.push({ Key: key, Value: value })
+    }
+    const taggingConfig = {
+      Tagging: {
+        TagSet: {
+          Tag: tagsList,
+        },
+      },
+    }
+    const encoder = new TextEncoder()
```
[the remark this answers is not in the document]
Comment (prakashsvmx) [reply src/internal/client.ts]: sure thank you,  We can revisit this later. 
i could not recollect why i added it. not sure if it was due to browser ( web browser ) support.

[the remark this answers is not in the document]
Comment (trim21) [reply src/internal/client.ts]: Yes, it's added in this PR: https://github.com/minio/minio-js/pull/955

[the remark this answers is not in the document]
Comment (prakashsvmx) [reply src/internal/client.ts]: :100: .  Yeah. We can look at supporting web browser environment post migration.

File: src/internal/client.ts [hunk 3]
```diff
@@ -1626,4 +1624,108 @@ export class TypedClient {
...
+
+      ...(objectName && { objectName: objectName }),
+    }
+
+    headers['Content-MD5'] = toMd5(payloadBuf)
+
+    await this.makeRequestAsyncOmit(requestOptions, payloadBuf)
+  }
+
+  async removeTagging({ bucketName, objectName, removeOpts }: RemoveTaggingParams): Promise<void> {
```
Comment (aldy505) [inline src/internal/client.ts:1664]: Should add `private` identifier to make sure this function is never called from outside of this class.

```suggestion
  private async removeTagging({ bucketName, objectName, removeOpts }: RemoveTaggingParams): Promise<void> {
```

File: src/internal/client.ts [hunk 4]
```diff
@@ -1626,4 +1624,108 @@ export class TypedClient {
 
     await this.makeRequestAsyncOmit({ method, bucketName, query, headers }, payload)
   }
+
+  async setTagging(taggingParams: PutTaggingParams): Promise<void> {
```
Comment (aldy505) [inline src/internal/client.ts:1628]: Should add `private` identifier to make sure this function is never called from outside of this class.

```suggestion
  private async setTagging(taggingParams: PutTaggingParams): Promise<void> {
```

File: examples/remove-bucket-tagging.js
Comment (trim21) [inline examples/remove-bucket-tagging.js:1]: mjs file ext？

**Review by kaankabalak — APPROVED** [review]
LGTM

File: src/minio.d.ts [hunk 5]
```diff
@@ -207,6 +207,9 @@ export class Client extends TypedClient {
   removeBucketTagging(bucketName: string, callback: NoResultCallback): void
   removeBucketTagging(bucketName: string): Promise<void>
 
+  setBucketVersioning(bucketName: string, versioningConfig: any, callback: NoResultCallback): void
```
Comment (trim21) [inline src/minio.d.ts:210]: why add these two methods?

File: src/minio.d.ts [hunk 6]
```diff
@@ -207,6 +207,9 @@ export class Client extends TypedClient {
   removeBucketTagging(bucketName: string, callback: NoResultCallback): void
```
Comment (trim21) [inline src/minio.d.ts:207]: should remove these 2 methods and `setBucketTagging`

Comment (prakashsvmx) [reply src/minio.d.ts]: Thank you. :+1:

File: src/minio.d.ts (hunk 5 above)
Comment (prakashsvmx) [reply src/minio.d.ts:204]: This would be cleaned up in a new PR as we decided to drop callback support midway.

**Review by kaankabalak — APPROVED** [review]

**Pull request merged**